### ISSUE TITLE
Blinking Cursor, but no input prompt (temporary fix)

### ISSUE THREAD

**Issue by micage** [jeremyramin/terminal-plus#402]
Hi,
a lot of people - myself included - seem to struggle with the latest version of Terminal-Plus.
There is only this blinking caret but no text input possible.
I tried the following procedure with success and it seems to be working as expected.  Hope this will help some of you, too:

In ~/.atom/packages/terminal-plus/package.json in the dependencies section
remove the commit id (#......) at the end of the pty.js entry:

"pty.js": "git+https://github.com/jeremyramin/pty.js.git#28f2667" becomes
"pty.js": "git+https://github.com/jeremyramin/pty.js.git"

cd ~/.atom/packages/terminal-plus/
npm install
apm rebuild

Restart Atom, this should be it ;)

**Comment by simonmittag** [comment]
this worked beautifully, thank you!

**Comment by mchljams** [comment]
I am also confirming that this fixed the issue for me. Thanks.

**Comment by pablocpantoja** [comment]
Work for me, thank you

**Comment by iamchathu** [comment]
When will this fix be added to release?

**Comment by suiteki** [comment]
It works! thanks!

**Comment by zzzgit** [comment]
![qq 20161128192607](https://cloud.githubusercontent.com/assets/1060733/20666750/c3e377c0-b5a0-11e6-8318-3b5db8ad960c.jpg)

**Comment by jonattanva** [comment]
thanks!

**Comment by insight1111** [comment]
I'm windows user.
I have errors similar zzzgit when `npm install`.

**Comment by ghost** [comment]
insight1111 have you tried updating node and npm?

**Comment by insight1111** [comment]
thank you, TacoCoder.
But I still failed install.
node version is 6.9.1 and npm version is 3.10.9.

**Comment by zzzgit** [comment]
now i have updated node to 6.9.1 and npm to 4.0.2
but i can't input into the panel still

**Comment by llong** [comment]
Mine seems to be mostly working after micage's fix but if I try to run 'Atom .' in the terminal I get this error "LSOpenURLsWithRole() failed for the application /Applications/Atom.app with error -10810."

**Comment by Linicks** [comment]
This fix worked for me: 
- Ubuntu, 16.04
- Atom, 1.12.6
- Node, 6.9.1
- NPM, 4.0.2

Thanks!

**Comment by ste-goldstein** [comment]
I was having same error as @zzzgit and @insight1111 but after upgrading node to 6.9.1 and npm to 4.02  the fix worked (Ubuntu 16.04).

**Comment by insight1111** [comment]
update npm 4.0.2 and atom updaet to 1.12.6.
It worked.Thank you!

**Comment by danrjohnson** [comment]
```$ npm install
npm ERR! unpack Could not read data for terminal-plus@0.14.5
npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/Cellar/node/6.0.0/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v6.0.0
npm ERR! npm  v3.8.6
npm ERR! code E404

npm ERR! 404 Registry returned 404 for GET on https://registry.npmjs.org/terminal-plus
npm ERR! 404
npm ERR! 404  'terminal-plus' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/danjohnson/.atom/packages/terminal-plus/npm-debug.log```

**Comment by josdima** [comment]
it worked. Ubuntu 16.04
Thanks.

**Comment by robrecord** [comment]
Please! Release this fix. Bugged me for weeks.

**Comment by blakekl** [comment]
Has anyone actually created a pull request for this? The fix worked for me.

[EDIT]
Just saw there is a pull request for this. It also fixes issue #362 
This issue seems to be a duplicate of that one.

**Comment by Somil91** [comment]
solution worked for me as well . Thanks

**Comment by j-burlison** [comment]
Solution worked for me! Thank you!!!

**Comment by skleeschulte** [comment]
I have the same issue. The fix does not work for me (Windows 7 x64; node-gyp working correctly).

**Comment by irbigturtle** [comment]
still experiencing this issue - thanks for the fix :) (OSX)

**Comment by UncleVic** [comment]
Sorry, maybe I'm stupid, but I can't do it.

macOS 10.12.1, Atom 1.12.6, node-gyp 3.4.0, node v7.2.0, npm v3.10.9

I've done that:

1. Close atom
2. Change the said line
3. Do the following commands
```
cd ~/.atom/packages/terminal-plus/
npm install
apm rebuild
```
4. Restart atom

But I see black screen and blincking cursor ;(
What am I doing wrong?

platformio-ide-terminal dosen't work too.

**Comment by prcolaco** [comment]
Worked like a charm for me. Thanks micage! :)

**Comment by redeemefy** [comment]
Did work for me as well.

**Comment by irbigturtle** [comment]
@UncleVic I would try re-installing terminal plus and doing it again. All of those version numbers look correct. Keep an eye on the results of each command. One might ask you for elevated privileges - maybe you missed it the first time around?

**Comment by UncleVic** [comment]
@irbigturtle, I tried re-installig terminal and Atom. [I can show](https://youtu.be/1wCGu7uqTjY).

**Comment by JordanRickman** [comment]
Worked for me as well. Thanks!

**Comment by UncleVic** [comment]
Can somebody help me?

**Comment by JordanRickman** [comment]
@zzzgit @insight1111 @UncleVic Have you installed the dependencies as described at https://github.com/nodejs/node-gyp#installation? I am not sure if this is required for normal usage, but the top of the terminal-plus README makes note of it. In particular, the dependencies are python 2.7, gcc, and make, so I can see those being needed to rebuild the package.

**Comment by UncleVic** [comment]
Yes, of course.
I can say more. Be honest, when terminal-plus was stoped work, I uninstalled it and install platformio-ide-terminal. It worked fine. But after #408 don't work both. May be not after #408, but I haven't used Atom for a long time before #408.

I've just re-installed Commandline tools for XCode, but nothing new... terminal-plus and platformio-ide-terminal don't work. :(

**Comment by quinob** [comment]
Fix worked ok . ubuntu 14.04 atom 1.12.6
Thank you !!

**Comment by tegument** [comment]
+1 fixed, still an issue with fresh install

Thanks!

**Comment by MichaelIT** [comment]
same as @zzzgit @insight1111 @UncleVic ,maybe this error only occur in mac OSX

**Comment by kenmorechalfant** [comment]
+1 fixed on Ubuntu 14.04

**Comment by UncleVic** [comment]
No, it isn't OSX.
I have home Mac and this patch works ok. But my MacBook... :(
And platformio-ide-terminal works too.

**Comment by yvuarajv** [comment]
it's not works in MAC, I think this works only for Ubuntu os environment. 
I tried in my mac it's not working.
Is there anyone who fixed this issue in OSX?
if yes, can you clarify it.

**Comment by UncleVic** [comment]
Please, watch [this](https://youtu.be/C9tedaDtHeY)
This is Atom 1.12.7

**Comment by iamchathu** [comment]
This works in MBP with MacOS Sierra

**Comment by unlimiworks** [comment]
### Update
It seems all these errors are exclusive to windows (winpty as pty.js' dependency). I believe this problem is is explained in [this issue](https://github.com/rprichard/winpty/issues/87) blaming VS/C++11 changes and the 2 commits in the answer fix it.
Is there any way for me to make it work the way things stand currently ?
______________________________
**Fix not working for me**

Win7 x64
atom 1.12.7
`apm-v` result :
```
apm 1.12.9
npm 3.10.5
python 2.7.12
visual studio 2015
```
here's `npm install` 's error / output
![node-gyp-1](https://cloud.githubusercontent.com/assets/10166333/21075563/6acc490c-bf0d-11e6-88aa-d58b437f4473.PNG)
![node-gyp-2](https://cloud.githubusercontent.com/assets/10166333/21075564/70960cb0-bf0d-11e6-94a4-15ce03dd3e95.PNG)

**Comment by irbigturtle** [comment]
@UncleVic @UnlimiWorks I'm Atom 1.12.6 OSX 10.12.1. Fix is working.

Vic - it seems like you have something else going on. Some conflict. Possibly platformio-ide-terminal? Something else that is interacting with terminal. Just a thought.

**Comment by UncleVic** [comment]
No, I have clear installation. I've removed prevision version and ~/.atom/.

**Comment by kdekooter** [comment]
`apm rebuild` resulted in complete removal of Atom from my machine. So no not really a good workaround.

**Comment by brucezlata** [comment]
works on Ubuntu. Thanks.
But when this issue would be fixed?

**Comment by PavanKu** [comment]
Worked for me too.. Thanks a lot.

**Comment by MichaelIT** [comment]
@UncleVic this error occur in my iMac and my MacBook Pro both😢

**Comment by pertrai1** [comment]
@micage thank you very much.

**Comment by alexshapalov** [comment]
It works! Thanks.

**Comment by shikelong** [comment]
It's no work in windows 10

**Comment by phivid** [comment]
Works fine on:

> Ubuntu 16.04.1 LTS
>  atom v6.5.0

**Comment by MatsMaker** [comment]
It's not helped

> OS Linux Mint 17.2
> Atom v1.12.6

**Comment by mujan5427** [comment]
@micage It works! Thank you

> macOS Sierra 10.12.1
> Atom 1.12.7

**Comment by UncleVic** [comment]
Wow!!! How???

**Comment by oguzbilgic** [comment]
This worked for me as well.

**Comment by websemantics** [comment]
Initially didn't work, until I updated `node.js` and `npm` to latest,
Thank you for this,

**Comment by UncleVic** [comment]
OS? Versions?

**Comment by andreyluiz** [comment]
Yeah, works for me as well. I didn't find a PR to fix this. Is there a PR?

**Comment by sparkleholic** [comment]
It works. Thanks a lot !

**Comment by matthewselby** [comment]
This worked, thanks!


> MacOS Sierra 10.12.2
> Atom 1.12.7

**Comment by Ephigenia** [comment]
Fixed it for me as well.

> MacOS Sierra 10.12.2
> Atom 1.12.7

**Comment by bonizz** [comment]
@UnlimiWorks For Windows, try https://github.com/jeremyramin/terminal-plus/issues/415

**Comment by hsnfirooz** [comment]
this worked, thanks :)

**Comment by superpig** [comment]
it worked in my MacOS 10.11.5, thinks!

**Comment by UncleVic** [comment]
I did it!!!
I removed Node and Atom completely and install Node again from brew.
Now terminal works, after fix of course.

**Comment by ELHoussineT** [comment]
Worked fine, but can the maker release a release solving this issue?

**Comment by unlimiworks** [comment]
@bonizz Thanks. I used plateformio-ide-terminal to get a working terminal-plus fork on windows without any hassle a while ago. Will dive a little bit deeper on the solutions there when i get the time

**Comment by kiranloot** [comment]
above solution did not work for me. spend way too much time trouble shooting moving on to something better.

**Comment by ghost** [comment]
Nice, it works.

**Comment by arphenlin** [comment]
hope this will be fixed in the next version

**Comment by mshiltonj** [comment]
Did not work for me. :-/

**Comment by naoey** [comment]
Wasn't working initially on macOS Sierra 10.12.2, worked like a charm after `brew upgrade node` to 7.4.0 👍.

**Comment by bennypowers** [comment]
This didn't work for me on sierra

**Comment by yuanzhuohao** [comment]
Work for me, Thank you!!!!!!!!

**Comment by andreyluiz** [comment]
Removing atom and node.js to make it work sounds like 'restart the machine to fix high processing usage'. It is not a good solution at all.

**Comment by pipewrk** [comment]
Ok, it wasn't working for me. Here's what I did for it to work:

````
cd ~/.atom/packages
rm -rf terminal-plus
git clone https://github.com/jeremyramin/terminal-plus
cd terminal-plus
npm install
apm rebuild
````
<img width="804" alt="screen shot 2017-02-02 at 10 07 23 am" src="https://cloud.githubusercontent.com/assets/780157/22534520/a1fc3ce2-e92f-11e6-9456-94becf4730a8.png">

**Comment by hanibash** [comment]
Everyone, I think you should try switching to [Termination](https://github.com/Fred-Barclay/Termination). Fork of terminal-plus that works great. Not part of an IDE. Active development.

**Comment by andreyluiz** [comment]
[Platformio IDE terminal](https://github.com/platformio/platformio-atom-ide-terminal) is also nice.

**Comment by prcolaco** [comment]
I'm using iTerm2 since this broke again after Sierra update. Quite nice this terminal and it has an option to auto open/hide with hotkey (alt + space) - have to setup in settings, not enabled by default...

**Comment by magbello157** [comment]
This fix worked perfectly for me on Ubuntu 16.04, Atom 1.13.1 x64, Node 6.9.5

**Comment by dzt** [comment]
Thanks! Worked like a charm :)

**Comment by utiq** [comment]
Great fix, thanks!

**Comment by NicholasPeretti** [comment]
Thanks! It works just fine!

**Comment by jagrio** [comment]
Thanks for the fix! Termination works great too!

For all those using linux (ubuntu 14.04) having issues with 
```bash
npm install
```
like myself, you should install both `npm` and  `nodejs-legacy`, if not already installed
```bash
sudo apt-get install npm nodejs-legacy
```
I found out thanks to this [post](http://stackoverflow.com/questions/21168141/cannot-install-packages-using-node-package-manager-in-ubuntu) also.

**Comment by jpecjo** [comment]
Awesome find and fix. Cheers!

**Comment by Kila2** [comment]
thanks work for me!

**Comment by nakaras** [comment]
it worked, Thank you.... !

**Comment by PrimeTimeTran** [comment]
Worked like a charm for me. Thanks bro!

**Comment by freebit** [comment]
Thanks!

**Comment by 2rohityadav** [comment]
now working for me. 
somebody help me out.. still i am getting just blank screen

**Comment by bennypowers** [comment]
Please refrain from "Thank you" or "Me too" comments. When you post them, the entire thread is notified. Use reactions instead.

**Comment by vinod827** [comment]
Yes, this solution is working me on Mac

**Comment by nlound** [comment]
Thank You!

**Comment by maratheameya3** [comment]
![untitled](https://user-images.githubusercontent.com/33648650/41638945-8bfafeca-7479-11e8-8bde-ce2667de440b.png)
This is the error that I get please advise. I am using Windows 10.

**Comment by MrAmbiG** [comment]
OS: windows 10
This Solution worked: NO/FALSE
Atom version:  1.39
node: v12.4.0
Many other terminal packages simply fail to install: atom-ide-terminal and others
Forks of this package behaves the same way: vk-terminal and others
I have been having this problem since months. Every month I check back and reinstall, repeat all the steps and when it fails, I simply go back to Microsoft Visual Studio Code with great disappointment even though VSC offers an overall better experience than Atom IDE, I will come back when atom itself has an terminal which works effortlessly.

**Comment by MikeyWand** [comment]
> Hi,
> a lot of people - myself included - seem to struggle with the latest version of Terminal-Plus.
> There is only this blinking caret but no text input possible.
> I tried the following procedure with success and it seems to be working as expected. Hope this will help some of you, too:
> 
> In ~/.atom/packages/terminal-plus/package.json in the dependencies section
> remove the commit id (#......) at the end of the pty.js entry:
> 
> "pty.js": "git+https://github.com/jeremyramin/pty.js.git#28f2667" becomes
> "pty.js": "git+https://github.com/jeremyramin/pty.js.git"
> 
> cd ~/.atom/packages/terminal-plus/
> npm install
> apm rebuild
> 
> Restart Atom, this should be it ;)

<img width="1114" alt="terminal-plus-bug-fix" src="https://user-images.githubusercontent.com/18725978/83176005-6ec9e680-a0e2-11ea-8df1-b7bb4d2ddc9b.png">

Hi, can someone explain why it thinks I'm offline? I'm not, and my work vpn is off.
